devito/dse/rewriters.py

```diff
@@ -260,26 +233,28 @@ def _eliminate_inter_stencil_redundancies(self, cluster, template, **kwargs):
            temp1 = 2.0*ti[x,y,z]
            temp2 = 3.0*ti[x,y,z+1]
         """
+        exprs = cluster.exprs
+
         # For more information about "aliases", refer to collect.__doc__
-        aliases = collect(cluster.exprs)
+        aliases = collect(exprs)
 
         # Redundancies will be stored in space-varying temporaries
-        graph = FlowGraph(cluster.exprs)
-        time_invariants = {v.rhs: graph.time_invariant(v) for v in graph.values()}
+        is_time_invariant = make_is_time_invariant(exprs)
+        time_invariants = {e.rhs: is_time_invariant(e) for e in exprs}
```

Comment:
fits in one line?

Reply:
it would, but I want `make_is_time_invariant` to be called once